nekoyume/Assets/_Scripts/UI/Widget/Popup/PetEnhancementPopup.cs

```diff
@@ -59,60 +62,112 @@ public class PetEnhancementPopup : PopupWidget
         [SerializeField]
         private TextMeshProUGUI soulStoneCostText;
 
+        [SerializeField]
+        private GameObject soulStoneNotEnoughObject;
+
         [SerializeField]
         private TextMeshProUGUI ncgCostText;
 
         [SerializeField]
         private GameObject ncgCostObject;
 
+        [SerializeField]
+        private GameObject ncgNotEnoughObject;
+
         [SerializeField]
         private GameObject costObject;
 
+        [SerializeField]
+        private Button plusButton;
+
+        [SerializeField]
+        private Button minusButton;
+
+        [SerializeField]
+        private Button soulStoneIconButton;
+
         private readonly List<IDisposable> _disposables = new();
+        private readonly ReactiveProperty<int> _enhancementCount = new();
 
         private const string LevelUpText = "LEVEL UP";
         private const string SummonText = "Summon";
 
         private int _targetLevel;
+        private int _sliderMax;
+        private int _sliderCurrentValue;
+        private bool _notEnoughBalance;
+        private PetSheet.Row _petRow;
 
         protected override void Awake()
         {
             base.Awake();
             closeButton.onClick.AddListener(() => Close());
+            LoadingHelper.PetEnhancement.Subscribe(id =>
+            {
+                buttonDisableObject.SetActive(_notEnoughBalance || id != 0);
+            }).AddTo(gameObject);
+            _enhancementCount.Subscribe(value =>
+            {
+                slider.ForceMove(value);
+            }).AddTo(gameObject);
```

Comment:
These streams are still in memory even if this popup is closed. I think it should be opened on `Show`, and disposed on `OnDisable`